yupiik-batch-runtime/src/main/java/io/yupiik/batch/runtime/batch/builder/BatchChain.java

```diff
@@ -249,7 +254,7 @@ private void await(final RunConfiguration configuration, final List<CompletableF
         }
         final var duration = configuration == null ? TimeUnit.MINUTES.toMillis(1) : configuration.maxBatchPromiseAwait;
         for (final var stage : new ArrayList<>(promises)) {
-            if (stage.isDone() || stage.isCompletedExceptionally()) {
+            if (stage.isDone() && (!configuration.failOnPromiseError || !stage.isCompletedExceptionally() )) {
```

Comment:
configuration can be null

Reply:
Use "true" value if configuration is null

Reply:
hmm, where's the magic which makes "configuration" not null when it is null? :thinking: